tests/integration/nm/vxlan_test.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+
+
+def test_create_and_remove_vxlan(eth1_up):
+    eth1_ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    vxlan_desired_state = {
+        VXLAN.CONFIG_SUBTREE: {
+            VXLAN.ID: 101,
+            VXLAN.BASE_IFACE: eth1_ifname,
+            VXLAN.REMOTE: '192.168.1.18',
+            VXLAN.DESTINATION_PORT: 4789,
+        }
+    }
+
+    with _vxlan_interface(vxlan_desired_state):
+
+        vxlan_current_state = _get_vxlan_current_state(vxlan_desired_state)
+        assert vxlan_desired_state == vxlan_current_state
+
+    assert not _get_vxlan_current_state(vxlan_desired_state)
+
+
+@pytest.mark.xfail(
+    strict=True,
+    reason=(
+        "Pending fix https://gitlab.freedesktop.org/NetworkManager/"
+        "NetworkManager/commit/28110ee3c1e997f837312cfc1c7808dc5c5c0fa3"
+    ),
+)
+def test_read_destionation_port_from_libnm(eth1_up):
+    eth1_ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    vxlan_desired_state = {
+        VXLAN.CONFIG_SUBTREE: {
+            VXLAN.ID: 101,
+            VXLAN.BASE_IFACE: eth1_ifname,
+            VXLAN.REMOTE: '192.168.1.18',
+            VXLAN.DESTINATION_PORT: 4789,
+        }
+    }
+
+    with _vxlan_interface(vxlan_desired_state):
+
+        nmdev = _get_vxlan_device(vxlan_desired_state)
+        assert (
+            nmdev.props.dst_port
+            == vxlan_desired_state[VXLAN.CONFIG_SUBTREE][
+                VXLAN.DESTINATION_PORT
+            ]
+        )
+
+
+@contextmanager
+def _vxlan_interface(state):
+    try:
+        _create_vxlan(state)
+        yield
+    finally:
+        _delete_vxlan(_get_vxlan_ifname(state))
+
+
+def _get_vxlan_device(vxlan_desired_state):
+    nm.nmclient.client(refresh=True)
+    ifname = _get_vxlan_ifname(vxlan_desired_state)
+    return nm.device.get_device_by_name(ifname)
+
+
+def _get_vxlan_current_state(vxlan_desired_state):
+    nmdev = _get_vxlan_device(vxlan_desired_state)
+    return nm.vxlan.get_info(nmdev) if nmdev else {}
+
+
+def _create_vxlan(vxlan_desired_state):
+    ifname = _get_vxlan_ifname(vxlan_desired_state)
+    con_setting = nm.connection.ConnectionSetting()
+    con_setting.create(
+        con_name=ifname,
+        iface_name=ifname,
+        iface_type=nm.nmclient.NM.SETTING_VXLAN_SETTING_NAME,
+    )
+    vxlan_setting = nm.vxlan.create_setting(
+        vxlan_desired_state, base_con_profile=None
+    )
+    ipv4_setting = nm.ipv4.create_setting({}, None)
+    ipv6_setting = nm.ipv6.create_setting({}, None)
+    with mainloop():
+        con_profile = nm.connection.ConnectionProfile()
+        con_profile.create(
+            (con_setting.setting, vxlan_setting, ipv4_setting, ipv6_setting)
+        )
+        con_profile.add(save_to_disk=False)
+        nm.device.activate(connection_id=ifname)
+
+
+def _delete_vxlan(devname):
+    nmdev = nm.device.get_device_by_name(devname)
+    with mainloop():
+        nm.device.deactivate(nmdev)
+        nm.device.delete(nmdev)
+
+
+def _get_vxlan_ifname(state):
```

Comment:
I am removing the 'get' preffix but keeping the function since we don't want to harcode the "eth1" interface name we get from the fixture and we are using that name as suffix of vxlan interface name.